packages/flutter/lib/src/widgets/table.dart

```diff
@@ -292,21 +292,34 @@ class _TableElement extends RenderObjectElement {
   @override
   void mount(Element? parent, dynamic newSlot) {
     super.mount(parent, newSlot);
-    _children = widget.children.map<_TableElementRow>((TableRow row) {
-      return _TableElementRow(
-        key: row.key,
-        children: row.children!.map<Element>((Widget child) {
-          assert(child != null);
-          return inflateWidget(child, null);
-        }).toList(growable: false),
-      );
-    }).toList(growable: false);
-    _updateRenderObjectChildren();
+    int index = 0;
+    final List<_TableElementRow> newChildren = <_TableElementRow>[];
+    for (int y = 0; y < widget.children.length; y++) {
+      final List<Element> elementChildren = <Element>[];
+      Element? previousChild;
+      for (int x = 0; widget.children[y].children != null && x < widget.children[y].children!.length; x++) {
+        assert(widget.children[y].children![x] != null);
+        final Element newChild = inflateWidget(widget.children[y].children![x], IndexedSlot<Element?>(index, previousChild));
+        elementChildren.add(newChild);
+        previousChild = newChild;
+        index++;
+      }
+      newChildren.add(_TableElementRow(
+        key: widget.children[y].key,
+        children: elementChildren,
+      ));
+    }
+    _children = newChildren;
   }
 
   @override
-  void insertRenderObjectChild(RenderObject child, dynamic slot) {
+  void insertRenderObjectChild(RenderObject child, IndexedSlot<Element?> slot) {
```

Comment:
It appears that you don't really need an IndexedSlot here. Wouldn't it be more convenient to just define a _TableSlot with an x and y property representing the row/column of the child?

Reply:
(If we do that, I am wondering whether we still need the TableCellParentData, which contains the same information and appears to only be used in removeRenderObjectChild)

Reply:
In this bug scenario, we must use IndexedSlot, we need the position info to decide where to insert the new RO.